task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,265 @@
+  results:
+    - description: Tekton task test output.
+      name: TEST_OUTPUT
+  params:
+    - description: Image URL.
+      name: image-url
+      type: string
+      default: ""
+    - description: Image digest to report findings for.
+      name: image-digest
+      type: string
+      default: ""
+    - name: FIND_UNICODE_CONTROL_GIT_URL
+      type: string
+      description: URL from repository to find unicode control
+      default: "https://github.com/siddhesh/find-unicode-control.git#c2accbfbba7553a8bc1ebd97089ae08ad8347e58"
+    - name: KFP_GIT_URL
+      type: string
+      description: URL from repository to download known false positives files
+      default: "https://gitlab.cee.redhat.com/osh/known-false-positives.git"
+    - default: ""
+      description: |
+        Name-Version-Release (NVR) of the scanned project.
+        It is used to find path exclusions (it is optional)
+      name: PROJECT_NVR
+      type: string
+    - default: "false"
+      description: |
+        Whether to record the excluded findings (defaults to false).
+        If `true`, the the excluded findings will be stored in `excluded-findings.json`.
+      name: RECORD_EXCLUDED
+      type: string
+  volumes:
+  steps:
+    - name: sast-unicode-check
+      image: quay.io/redhat-appstudio/konflux-test:v1.4.6@sha256:5f298d8d990dfa82023e50029b71b08e19c3c9cedb181dfc4bc86c9ecad8700c
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+      env:
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: FIND_UNICODE_CONTROL_GIT_URL
+          value: $(params.FIND_UNICODE_CONTROL_GIT_URL)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: SOURCE_CODE_DIR
+          value: $(workspaces.workspace.path)
+      script: |
+        #!/usr/bin/env bash
+        set -exuo pipefail
+        . /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        SCAN_PROP=""
+
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        curl -sS https://certs.corp.redhat.com/certs/2015-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2015-RH-IT-Root-CA.crt
+        curl -sS https://certs.corp.redhat.com/certs/2022-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2022-IT-Root-CA.pem
+        update-ca-trust
+
+        # Clone the source code from upstream repo
+        clone_unicontrol()
+        {
+            GIT_URL=$(echo ${FIND_UNICODE_CONTROL_GIT_URL} | awk -F'#' '{print $1}')
+            REV=$(echo ${FIND_UNICODE_CONTROL_GIT_URL} | awk -F'#' '{print $2}')
+
+            # Clone find-unicode-control repository
+            git clone --depth=1 "${GIT_URL}" cloned >&2
```

Comment:
I use `cloned`, because I find some other tasks use it, if it is unusual, I will  use implicit name `find-unicode-control`.

 `git clone --depth=1 "${GIT_URL}" cloned >&2`, `"${GIT_URL}"` is only a git repository url, not container REV

Reply:
Thanks for the update!  I meant let's use `find-unicode-control` as the implicit name but it needs to be explicitly specified in the `git clone` command (so that it works for arbitrary URLs):
```
git clone "${GIT_URL}" find-unicode-control
```

The problem with `--depth=1` has not been fixed.  Currently this works only because c2accbfbba7553a8bc1ebd97089ae08ad8347e58 is the last commit in the default upstream branch.  Once upstream pushes one more commit to the default branch, the `git checkout` command will start to fail (with the default `FIND_UNICODE_CONTROL_GIT_URL`).  You can try this out by setting `https://github.com/siddhesh/find-unicode-control.git#8599acba673ca4e300c818315bc79edea3509f93` as `FIND_UNICODE_CONTROL_GIT_URL`.  It will break if `--depth=1` is used.

Reply:
Yes, you are right, after test, need remove `--depth=1`
```
chhan@chhan-mac build-definitions % FIND_UNICODE_CONTROL_GIT_URL="https://github.com/siddhesh/find-unicode-control.git#8599acba673ca4e300c818315bc79edea3509f93"
chhan@chhan-mac build-definitions % GIT_URL=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $1}')
chhan@chhan-mac build-definitions % echo $GIT_URL
https://github.com/siddhesh/find-unicode-control.git
chhan@chhan-mac build-definitions % REV=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $2}')
chhan@chhan-mac build-definitions % echo $REV
8599acba673ca4e300c818315bc79edea3509f93
chhan@chhan-mac build-definitions % git clone --depth=1 "${GIT_URL}" find-unicode-control    
Cloning into 'find-unicode-control'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (11/11), done.
remote: Total 14 (delta 0), reused 9 (delta 0), pack-reused 0 (from 0)
Receiving objects: 100% (14/14), 7.69 KiB | 1.54 MiB/s, done.
chhan@chhan-mac build-definitions % git -C ./find-unicode-control/ checkout --quiet "${REV}"
fatal: reference is not a tree: 8599acba673ca4e300c818315bc79edea3509f93
chhan@chhan-mac build-definitions % 


chhan@chhan-mac build-definitions % git clone "${GIT_URL}" find-unicode-control  
Cloning into 'find-unicode-control'...
remote: Enumerating objects: 64, done.
remote: Counting objects: 100% (64/64), done.
remote: Compressing objects: 100% (43/43), done.
remote: Total 64 (delta 29), reused 41 (delta 15), pack-reused 0 (from 0)
Receiving objects: 100% (64/64), 14.68 KiB | 224.00 KiB/s, done.
Resolving deltas: 100% (29/29), done.
chhan@chhan-mac build-definitions % git -C ./find-unicode-control/ checkout --quiet "${REV}"
chhan@chhan-mac build-definitions % 
```